File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> ans = new ArrayList<>();
        
        // Mark elements as negative to indicate their presence
        for (int c : nums) {
            int index = Math.abs(c) - 1;
            nums[index] = -Math.abs(nums[index]);
        }
        
        // Find the indices which are positive, these indices + 1 are the missing numbers
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                ans.add(i + 1);
            }
        }
        
        return ans;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.go
func findDisappearedNumbers(nums []int) []int {
    for i := range nums {
        if nums[ abs(nums[i]) - 1] > 0 {
            nums[ abs(nums[i]) - 1] *= -1
        }
    }

    res := []int{}
    for i := range nums {
        if nums[i] > 0 {
            res = append(res , i  + 1)
        }
    }
    return res
}

func abs(num int) int {
    if num < 0 {
        return -num
    }
    return num
}